.travis.yml

```diff
@@ -11,18 +11,42 @@ cache:
     - "bower_components"
 
 script:
+  - zip -r latest *
+  - mkdir -p deploy_upload
+  - mv latest.zip deploy_upload/latest.zip
   - npm run lint
   - npm run babel
   - npm run webpack
   - npm run test
   - npm run esdoc
 
 deploy:
-  provider: pages
+- provider: pages
   skip-cleanup: true
   github-token: ${GH_TOKEN}
   keep-history: true
   on:
     branch: master
   local-dir: docs
   target-branch: gh-pages
+- provider: s3
+  access_key_id: $AWS_ACCESS_KEY
+  secret_access_key: $AWS_SECRET_KEY
+  local_dir: deploy_upload
+  skip_cleanup: true
+  on:
+    branch: staging
+    repo: GraceCovenantChurch/gcc_website
+  bucket: s3.web.staging
+  region: us-east-1
+- provider: codedeploy
+  access_key_id: $AWS_ACCESS_KEY
+  secret_access_key: $AWS_SECRET_KEY
+  bucket: s3.web.staging
+  key: latest.zip
+  bundle_type: zip
+  application: gcc.web.staging
+  deployment_group: gccweb
+  region: us-east-1
+  on:
+    branch: staging
```

Comment:
Does this need `repo: GraceCovenantChurch/gcc_website` as well?

Reply:
Nope. `deployment_group` is a keyword in CodeDeploy.